src/lightning/app/cli/connect/node.py

```diff
@@ -0,0 +1,177 @@
+    if sys.platform == "win32":
+        _error_and_exit("Node connection isn't supported on windows. Open an issue on Github.")
+
+    if "lightning.ai" in CLOUD_PROXY_HOST:
+        _error_and_exit("Node connection isn't publicly available. Open an issue on Github.")
+
+    with Live(Spinner("point", text=Text("pending...", style="white")), transient=True) as live:
+        # run network creation in the background
+        out = subprocess.run(CMD_CREATE_NETWORK, shell=True, capture_output=True)
+        error = out.stderr
+        if error:
+            if "already exists" not in str(error):
+                live.stop()
+                rich.print(f"[red]Failed[/red]: network creation failed with error: {str(error)}")
+                return
+
+        # if code server is already running, ignore.
+        # If not, but container exists, remove it and run. Otherwise, run.
+        out = subprocess.run(
+            f"docker ps -q -f name={CODE_SERVER_CONTAINER}", shell=True, check=True, capture_output=True
+        )
+        if out.stdout:
+            pass
+        else:
+            out = subprocess.run(
+                f"docker container ls -aq -f name={CODE_SERVER_CONTAINER}", shell=True, check=True, capture_output=True
+            )
+            if out.stdout:
+                subprocess.run(f"docker rm -f {CODE_SERVER_CONTAINER}", shell=True, check=True)
+            else:
+                live.update(Spinner("point", text=Text("pulling code server image", style="white")))
+                out = subprocess.run(f"docker pull {CODE_SERVER_IMAGE}", shell=True, check=True, capture_output=True)
+                error = out.stderr
+                if error:
+                    live.stop()
+                    rich.print(f"[red]Failed[/red]: code server image pull failed with error: {str(error)}")
+                    return
+            cmd = get_code_server_docker_command()
+            live.update(Spinner("point", text=Text("running code server", style="white")))
+            _ = subprocess.Popen(shlex.split(cmd), stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+
+        # if lightning daemon is already running, ignore.
+        # If not, but container exists, remove it and run. Otherwise, run.
+        out = subprocess.run(
+            f"docker ps -q -f name={LIGHTNING_DAEMON_CONTAINER}", shell=True, check=True, capture_output=True
+        )
+        if out.stdout:
+            pass
+        else:
+            out = subprocess.run(
+                f"docker container ls -aq -f name={LIGHTNING_DAEMON_CONTAINER}",
+                shell=True,
+                check=True,
+                capture_output=True,
+            )
+            if out.stdout:
+                subprocess.run(f"docker rm -f {LIGHTNING_DAEMON_CONTAINER}", shell=True, check=True)
+            else:
+                live.update(Spinner("point", text=Text("pulling lightning daemon image", style="white")))
+                out = subprocess.run(
```

Comment:
Suggestion: Could you refactor subprocess.run into a function.

```python
out = system(f"docker pull {LIGHTNING_DAEMON_IMAGE}")

```

Reply:
Not a great abstraction because we are using different arguments of `run` function in differnt places. May be I could have different high level functions for each task